packages/blast-core/lib/cudos-contract.js

```diff
@@ -10,81 +10,172 @@ const {
-    this.#signer = signer
-    this.#contractAddress = deployedContractAddress
+  constructor(contractLabel) {
+    this.#label = contractLabel
     this.#gasPrice = GasPrice.fromString(getGasPrice())
+  }
 
-    if (deployedContractAddress === null) {
-      this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractLabel}.wasm`)
-      if (!fs.existsSync(this.#wasmPath)) {
-        throw new BlastError(`Contract with label ${contractLabel} was not found, did you compile it?`)
-      }
+  static constructLocal(label) {
+    const wasmPath = path.join(getProjectRootPath(), `artifacts/${label}.wasm`)
+    if (!fs.existsSync(wasmPath)) {
+      throw new BlastError(`Contract with label ${label} was not found. Only compiled contracts can be accepted`)
     }
+    const cudosContract = new CudosContract(label)
+    cudosContract.#wasmPath = wasmPath
+    return cudosContract
+  }
+
+  static async constructUploaded(codeId) {
+    const codeDetails = await getCodeDetails(getNetwork(process.env.BLAST_NETWORK), codeId)
+    const cudosContract = new CudosContract()
+    cudosContract.#codeId = codeId
+    cudosContract.#creator = codeDetails.creator
+    return cudosContract
+  }
+
+  static async constructDeployed(contractAddress) {
+    const contractInfo = await getContractInfo(getNetwork(process.env.BLAST_NETWORK), contractAddress)
+    const cudosContract = new CudosContract(contractInfo.label)
+    cudosContract.#codeId = contractInfo.codeId
+    cudosContract.#contractAddress = contractInfo.address
+    cudosContract.#creator = contractInfo.creator
+    return cudosContract
   }
 
-  async deploy(initMsg, options = {}) {
-    if (options.signer) {
-      this.#signer = options.signer
+  // Uploads the contract's code to the network
+  async uploadCode(options = { signer: null }) {
+    if (this.#isUploaded()) {
+      throw new BlastError('Cannot upload contract that is already uploaded')
     }
-    if (!options.signer && this.#contractAddress === null) {
-      this.#signer = await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    options.signer = options.signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    const uploadTx = await this.#uploadContract(options.signer)
+    this.#codeId = uploadTx.codeId
+    this.#creator = options.signer
+    return uploadTx
+  }
+
+  // Instantiates uploaded code without assigning the new contract to current contract object instance
+  async instantiate(msg, options = {
+    signer: null, label: null, funds: null
+  }) {
+    if (!this.#isUploaded()) {
+      throw new BlastError('Cannot instantiate contract that is not uploaded. ' +
+        'Contract\'s code must exist on the network before instantiating')
     }
-    if (!options.label) {
-      options.label = this.#contractLabel
+    options.signer = options.signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    options.label = options.label ?? this.#label
```

Comment:
fix the issue here as discussed

Reply:
I will only leave` this.#label `in the contract object so the user is able to `getContractFromAddress()` and check the label by `getLabel() `function. (The purpose of having `label` will be similar to `creator`) Let me know if you have other suggestions